resources/automatev2.rb

```diff
@@ -0,0 +1,82 @@
+#
+# Cookbook:: chef-ingredient
+# Resource:: automatev2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource_name 'chef_automatev2'
+
+property :channel, Symbol, default: :current
+property :version, [String, Symbol], default: :latest
+property :config, String, required: true
+property :accept_license, [TrueClass, FalseClass], default: false
+
+action :create do
+  execute "curl https://packages.chef.io/files/#{new_resource.channel}/#{new_resource.version}/chef-automate-cli/chef-automate_linux_amd64.zip | gunzip - > chef-automate && chmod +x chef-automate" do
+    cwd '/usr/local/bin'
+    creates '/usr/loca/bin/chef-automate'
+    not_if { FileTest.file?('/usr/bin/chef-automate') }
+  end
+
+  ## TODO: add dependancy on sysctl cookbook unless chef-client v14.0+
```

Comment:
Apologies if this has already been discussed, but given this statement shouldn't the metadata.rb and travis matrix be updated to require 14+?

Alternatively, could we stick all of these in a function and guard them with a `respond_to?`

Reply:
only this one resource due to the use of the sysctl resource introduced would require chef-client 14+. I believe default behaviour now is to use built in resource if there is a conflict. So we may be able to simply add depends on sysctl cookbook in metadata to maintain compatibility with chef-client 13 since it's not EOL yet. Have to verify that assumption first.

Reply:
or would it be better to raise an error if they are not on chef-client 14+ since those sysctl settings need to be set for the automate preflight check to pass.

Reply:
I think we can fairly assume Chef 14+ at this point